### DISCUSSION TITLE
Changing the encoding of all elements in a DicomDataset

### DISCUSSION THREAD

**Discussion by TretMine** [fo-dicom/fo-dicom#1498] [Q&A]
Hi, 

I have a data set which is encoded with "ISO_IR 192". 
How can I change the complete data set to "ISO_IR 100"?

Is there a ready to use function or should I run through all encoding based tags and convert them manually?

Regards, 
TretMine.

**Comment by gofal** [comment]
When you open a DICOM file, then fo-dicom parses the values into System.String (unicode). If you then just call `dataset.AddOrUpdate(DicomTags.SpecificCharacterSet, "ISO_IR 100");` and then you store the dataset to disk. This will update all the String elements.
